### PR TITLE
[SA-1282] Fix: adding force orientation support to SideMenuController

### PR DESCRIPTION
**Pull request by dilumdesilva**
## What's changed
As this view controller is the rootViewController for most of the viewControllers in the Hectre app, SideMenuController has been modified to support force orientations via this change.

podspec version has been updated to `0.24` => `0.2.5` 

## Deployment notes
- As this pod was cloned to Hectre organisation app `Podfile` requires path changes
- Update app pod and local pods for SideMenuController using `pod update SideMenuController`

### CODE REVIEW

File: Source/SideMenuController.swift [hunk 1]
```diff
@@ -145,6 +145,19 @@ public extension SideMenuController {
...
+    Sets the supported orientation for the view
+
+    - parameter orientation: required orientation which is a type of UIInterfaceOrientationMask 
+    */
+    public func restrictOrientation(_ orientation: UIInterfaceOrientationMask) {
+        orientationLock = orientation
+    }
+    
+    open override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return orientationLock ?? .all
```
Comment (ariburgos) [inline Source/SideMenuController.swift]: Is this optional required? Looks like orientationLock is not optional

File: Source/SideMenuController.swift [hunk 2]
```diff
@@ -145,6 +145,19 @@ public extension SideMenuController {
 			prepare(centerControllerForContainment: controller, hideMenuButton: false)
 		}
 	}
+
+    /**
```
Comment (ariburgos) [inline Source/SideMenuController.swift:35]: Are these line properly indented or is it just GitHub?

File: Source/SideMenuController.swift [hunk 3]
```diff
@@ -145,6 +145,19 @@ public extension SideMenuController {
...
+    Sets the supported orientation for the view
+
+    - parameter orientation: required orientation which is a type of UIInterfaceOrientationMask 
+    */
+    public func restrictOrientation(_ orientation: UIInterfaceOrientationMask) {
+        orientationLock = orientation
+    }
+    
+    open override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
```
Comment (ariburgos) [inline Source/SideMenuController.swift]: This should go outside the scope of the` // MARK: - Public methods -` (line 32).

Add it into the main class or create a new extension to handle the orientation.

File: Source/SideMenuController.swift [hunk 4]
```diff
@@ -180,6 +193,8 @@ open class SideMenuController: UIViewController, UIGestureRecognizerDelegate {
 
 	/// Used for iOS 13+
 	var statusBarHidden = false
+    
+    private var orientationLock = UIInterfaceOrientationMask.all
```
Comment (ariburgos) [inline Source/SideMenuController.swift:193]: indentation?

**Review by ariburgos — CHANGES_REQUESTED** [review]

File: Source/SideMenuController.swift (hunk 1 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift]: Yes, I will remove the optional handling.

File: Source/SideMenuController.swift (hunk 2 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift:35]: Let me check on this.

File: Source/SideMenuController.swift (hunk 1 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift]: Fixed

File: Source/SideMenuController.swift (hunk 2 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift:35]: Indentation was fixed for the entire file

File: Source/SideMenuController.swift (hunk 4 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift:193]: Fixed for the entire file

File: Source/SideMenuController.swift (hunk 3 above)
Comment (dilumdesilva) [reply Source/SideMenuController.swift]: Moved it into the class

**Review by ariburgos — APPROVED** [review]
Looking great 🚀

**Pull request merged**